enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,454 @@
+  conformant as possible - that they pass most Kubernetes and
+  OpenShift conformance e2e tests. Fully document any tests that must
+  be skipped due to differences from a traditional deployment.
+
+### Non-Goals
+
+* Single node clusters supportable for production or critical task
+  workloads.  The cluster profile described in this enhancement
+  proposal is not intended to support any form of production cluster
+  deployment. Single node clusters for non-developer use cases, such
+  as for edge computing, would require distinct cluster profiles with
+  independent selection and configuration of components specific to
+  those use cases.
+* A new CRC installer. Although it should somewhat simplify the SNC
+  tooling used for CRC, this enhancement proposal does not change the
+  basic approach to installing CRC or producing a VM image for the CRC
+  installer. By making the definition of the content and configuration
+  of the components in a single node developer cluster more orthogonal
+  to the tooling that deploys the cluster, this enhancement could
+  enable distinct future enhancements to improve the build process
+  used by CRC.
+* A new OpenShift installer. This enhancement proposal does not add a
+  new OpenShift installer or involve changes, beyond the ability to
+  specify the cluster profile, to the standard OpenShift installer
+  currently used to build CRC installation images. Evolution of the
+  existing installer, SNC tooling and other CRC components is
+  expected.
+* Functional changes for OpenShift. This enhancement proposal does not
+  result in any functional changes to clusters that do not utilize the
+  single-node-developer cluster profile.
+
+## Proposal
+
+The single-node-developer cluster profile will be implemented in two
+phases. The first phase introduces the new cluster profile, and
+enables some simplification of the SNC image build process by
+utilizing the profile to control which components are included in the
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
+node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer=true
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* most operators running in the openshift-monitoring namespace
+* most operators running in the opensift-insights namespace
+* machine-api-operator
+* Machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
```

Comment:
From the current design, I would also exclude:

- openshift-cluster-node-tuning-operator
- openshift-multus

Reply:
Makes sense. I'll check with the team to see if they are aware of any reasons not to exclude these.

Reply:
Added text below.